cryptographic_estimators/MREstimator/mr_problem.py

```diff
@@ -0,0 +1,242 @@
+
+    @property
+    def theta(self):
+        """
+        Returns the value of `theta`
+
+        """
+        return self._theta
+
+    @theta.setter
+    def theta(self, value: float):
+        """
+        Sets the value of `theta`
+
+        """
+        self._theta = value
+
+    def expected_number_solutions(self):
+        """
+        Return the logarithm of the expected number of existing solutions to the problem
+
+        """
+        q, m, n, k, r = self.get_problem_parameters()
+        if k + 1 <= (m - r) * (n - r):
+            return 0
+        else:
+            return (k + 1 - (m - r) * (n - r)) * log2(q)
+
+    def order_of_the_field(self):
+        """
+        Return the order of the field
+
+        TESTS::
+
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: MRP = MRProblem(q=7, m=9, n=10, k=15, r=4)
+            sage: MRP.order_of_the_field()
+            7
+        """
+        return self.parameters[MR_FIELD_SIZE]
+
+    def nrows(self):
+        """"
+        Return the number of rows of the input matrix
+
+        TESTS::
+
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: MRP = MRProblem(q=7, m=9, n=10, k=15, r=4)
+            sage: MRP.nrows()
+            9
+        """
+        return self.parameters[MR_NUMBER_OF_ROWS]
+
+    def ncolumns(self):
+        """"
+        Return the number of columns of the input matrix
+
+        TESTS::
+
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: MRP = MRProblem(q=7, m=9, n=10, k=15, r=4)
+            sage: MRP.ncolumns()
+            10
+        """
+        return self.parameters[MR_NUMBER_OF_COLUMNS]
+
+    def length_solution_vector(self):
+        """"
+        Return the length of the solution vector
+
+        TESTS::
+
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: MRP = MRProblem(q=7, m=9, n=10, k=15, r=4)
+            sage: MRP.length_solution_vector()
+            15
+        """
+        return self.parameters[MR_LENGTH_SOLUTION_VECTOR]
+
+    def nmatrices(self):
+        """"
+        Return the number of input matrices
+
+        TESTS::
+
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: MRP = MRProblem(q=7, m=9, n=10, k=15, r=4)
+            sage: MRP.nmatrices()
+            16
+        """
+        return self.length_solution_vector() +  1
+
+    def target_rank(self):
+        """"
+        Return the target rank
+
+        TESTS::
+
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: MRP = MRProblem(q=7, m=9, n=10, k=15, r=4)
+            sage: MRP.target_rank()
+            4
+        """
+        return self.parameters[MR_TARGET_RANK]
+
+    def get_problem_parameters(self):
```

Comment:
there is a similar function defined in the BaseProblem class called get_parameters(). would it make sense to reuse it or if it doesn't suit your purpose exactly override it?